src/hooks/use-read-resource.ts

```diff
@@ -49,27 +50,29 @@ type UseReadResourceResult<T> = {
  * </pre>
  *
  */
-export function useReadResource<T>(resource: Resource<T> | string): UseReadResourceResult<T> {
+export function useReadResource<T>(resource: ResourceLike<T>): UseReadResourceResult<T> {
 
   const kettingContext = useContext(getKettingContext());
-  let res: Resource;
-  if (typeof resource === 'string') {
-    if (!kettingContext.client) {
-      throw new Error('To use useReadResource with a string argument, you must have a <KettingProvider> component set up');
-    }
-    res = kettingContext.client.go(resource);
-  } else {
-    res = resource;
-  }
 
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState<Error|null>(null);
   const [resourceState, setResourceState] = useState<ResourceState<T>>();
+  const [res, setRes] = useState<Resource<T>>();
```

Comment:
this is stating `res` is going to be typed as a Resource generic, which itself could of many kinds? Hence the `<T>`?

Reply:
For resources Resource<X> always means , that X refers to the type that you would get when you call `.get()` and the type that it requires when you call `.put()`.

So in our application it might be Resource<Video> or Resource<Category>. It's just a placeholder for something else.